### DISCUSSION TITLE
Prettier and PHPStan

### DISCUSSION THREAD

**Discussion by janschoenherr** [wielebenwir/commonsbooking#1685] [Ideas]
Hi,

have you considerer using Prettier and PHPStan. Both might make it easier for new contributors to find their way around the code base.

I could create a PR for both tools.

Regards
Jan

**Comment by hansmorb** [comment]
I think PHPStan is already in use @datengraben ?

**Comment by datengraben** [comment]
@hansmorb is right, @janschoenherr I already started with running the initial levels of phpstan (in this non-PR-commit https://github.com/wielebenwir/commonsbooking/commit/f555c18a8eff65187f9d87a4c631b20d8c079840) and fixing findings in the repo code with this PR https://github.com/wielebenwir/commonsbooking/pull/1654 and suppose you did the same with https://github.com/wielebenwir/commonsbooking/pull/1686, since there was a small overlap haha.

I hesitated because of the autloading issue, but after PR https://github.com/wielebenwir/commonsbooking/pull/1656 is merged now, onboarding of new contributors should have one less obstacle to overcome and it should be possible to start all of the tooling (phpstan/phpunit/phpcs) right after the initial repo setup. For every new/old contributor :)

I also stumbled upon this [Github Action](https://github.com/staabm/annotate-pull-request-from-checkstyle) that adds PR-Annotations via an Github-Action using the Checks-API and thought about integrating it into our github actions.

EDIT: PR-Link to phpstan config

**Reply by janschoenherr** [reply]
Great! No, I just used PhpStorm to find those issues 🙂 But if you can merge into the main branch, I would contribute to increasing the level to 2 or 3 😉

By the way, wouldn't it be good to move to GitFlow with a develop and a main branch? The develop branch as the default branch.

**Reply by datengraben** [reply]
Level 1 is no checked in here https://github.com/wielebenwir/commonsbooking/pull/1692/

**Reply by datengraben** [reply]
@janschoenherr level 1 is now checked via github action. https://github.com/wielebenwir/commonsbooking/pull/1707

**Comment by datengraben** [comment]
@janschoenherr and I know prettier, but did not check its capabillities for PHP code.

@janschoenherr I would be interested in pull request, but would like to have two seperate. One for phpstan and one for prettier.
Because at least for prettier I would prefer to discuss advantages of keeping phpcs first. Since phpcs seems to be widely prevalent in Wordpress-Plugins and has active development of the Wordpress-Styles. 

But till now, I also didn't check for any prettier php toolings. Maybe I will be suprised.

**Reply by janschoenherr** [reply]
We use https://github.com/prettier/plugin-php for the PHP code. I never used phpcs before, but that isn't a code formatter, right? I will do a PR for prettier and you can check if you like the results.

**Reply by janschoenherr** [reply]
I added a PR: https://github.com/wielebenwir/commonsbooking/pull/1687

**Reply by datengraben** [reply]
> We use https://github.com/prettier/plugin-php for the PHP code. I never used phpcs before, but that isn't a code formatter, right? I will do a PR for prettier and you can check if you like the results.

I only know phpcs (the linter) and phpcbf (the formatter) from the wordpress ecosystem. And it's heavy realiance on it (AFAIK) in the past.
